ProGetAutomation/Functions/Set-ProGetAsset.ps1

```diff
@@ -5,52 +5,69 @@ function Set-ProGetAsset
 
         .EXAMPLE
-        Set-ProGetAsset -Session $session -Path 'subdir/exampleAsset.txt' -DirectoryName 'assetDirectory' -FilePath 'path/to/file.txt'
+        Set-ProGetAsset -Session $session -DirectoryName 'assetDirectory'-Path 'subdir/exampleAsset.txt' -FilePath 'path/to/file.txt'
 
-        Example of adding an asset located on the machine at `path/to/file.txt` to ProGet in the `assetDirectory/subdir` folder. If `assetDirectory` is not created it will throw an error. If subdir is not created it will create the folder.
+        Example of publishing a file located at `path/to/file.txt` to ProGet in the `assetDirectory/subdir` folder. If `assetDirectory` is not created it will throw an error. If subdir is not created it will create the folder.
         
         .EXAMPLE
-        Set-ProGetAsset -Session $session -Path 'exampleAsset.txt' -Directory 'assetDirectory' -Path 'path/to/file.txt'
+        Set-ProGetAsset -Session $session -Directory 'assetDirectory' -Path 'exampleAsset.txt' -Body $bodyContent
 
-        Example of adding an asset located on the machine at `path/to/file.txt` to ProGet in the `assetDirectory` folder. If `assetDirectory` is not created it will throw an error. If subdir is not created it will create the folder.
+        Example of publishing content contained in the $bodyContent variable to ProGet in the `assetDirectory` folder.
     #>
+    [CmdletBinding()]
     param(
         [Parameter(Mandatory = $true)]
         [Object]
         # A session object that represents the ProGet instance to use. Use the `New-ProGetSession` function to create session objects.
         $Session,
-        
+
         [Parameter(Mandatory = $true)]
         [string]
-        # The name of a valid directory to upload the desired asset in ProGet. If no root directories exist, use the `New-ProGetFeed` with parameter `-Type 'Asset'` to create a new directory in the ProGet assets page.
-        $DirectoryName,        
-        
+        # The name of a valid root asset directory in ProGet. If no root directories exist, use the `New-ProGetFeed` with parameter `-Type 'Asset'` to create a new asset directory.
+        $DirectoryName,
+
         [Parameter(Mandatory = $true)]
         [string]
-        # Desired path of the asset that will be uploaded. Any directories that do not exist will be created automatically.
+        # The path where the asset will be published. Any directories that do not exist will be created automatically.
         $Path,
 
-        [Parameter(Mandatory = $true)]
+        [Parameter(Mandatory = $true, ParameterSetName = 'ByFile')]
         [string]
-        # The Relative Path of the file to be uploaded. 
-        $FilePath
+        # The relative path of a file to be published as an asset.
+        $FilePath,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'ByContent')]
+        [string]
+        # The content to be published as an asset.
+        $Body

```

Comment:
Rename to `$Value` (that's the parameter name the `Add-Content` and `Set-Content` cmdlets use).